core/model/modx/filters/modoutputfilter.class.php

```diff
@@ -594,6 +594,19 @@ public function filter(&$element) {
                             $output = urldecode($output);
                             break;
 
+                        case 'dirname':
+                            $output = pathinfo($output)['dirname'];
+                            break;
+                        case 'basename':
+                            $output = pathinfo($output)['basename'];
+                            break;
+                        case 'filename':
+                            $output = pathinfo($output)['filename'];
+                            break;
+                        case 'extension':
+                            $output = pathinfo($output)['extension'];
```

Comment:
In manual said that if a file does not have an extension value won't be returned. It should be additionally checked to avoid warnings or notices.
http://php.net/manual/en/function.pathinfo.php
Also, please read the note about files that starts with dot (`.`). Filename also can be empty.

Reply:
Hi, @Alroniks. Thanks a lot for code review. Now i get an idea, how to improve a little this PR.

About warning and notices. The pathinfo() function doesn't produce any warning or notices. Even if empty value will be given, this function will return empty "basename" and empty "filename". I think this behavior will not change, since function has not changed since version 5.2.

I can make check for empty output and return something: "basename not found" or "filename empty". But i think more reasonable return same as the function returns, because modifiers are only a wrappers for this function.

Any suggestions or idea?

Reply:
It is not correct.
```
php > $path = 'filename';
php > print_r(pathinfo($path));
Array
(
    [dirname] => .
    [basename] => filename
    [filename] => filename
)
```
Here `extension` key does not exist in the array but you try get array item with this key
```
$output = pathinfo($output)['extension'];
```
PHP will throw Notice that index not found.

Reply:
Sorry for my inattention (i have tested pathinfo function and forgot about array dereferencing).

I have update this PR with code, which completly lies on pathinfo() and now it works without any warnings.